src/server/scene/application_session.cpp

```diff
@@ -89,9 +89,9 @@ mf::SurfaceId ms::ApplicationSession::next_id()
     return mf::SurfaceId(next_surface_id.fetch_add(1));
 }
 
-mf::SurfaceId ms::ApplicationSession::create_surface(
+auto ms::ApplicationSession::create_surface(
     SurfaceCreationParameters const& the_params,
-    std::shared_ptr<mf::EventSink> const& surface_sink)
+    std::shared_ptr<mf::EventSink> const& surface_sink) -> std::shared_ptr<scene::Surface>
```

Comment:
One of the advantages of post-fixing the return type is that the namespace is known and doesn't need restating.